modules/nexverseBidAdapter.js

```diff
@@ -0,0 +1,447 @@
+      }
+
+      if (params.length > 0) {
+        syncUrl += '?' + params.join('&');
+      }
+
+      syncs.push({
+        type: 'iframe',
+        url: syncUrl,
+      });
+    }
+
+    return syncs;
+  },
+};
+
+/**
+ * Builds the OpenRTB 2.5 request payload.
+ *
+ * @param {Object} bid - The bid request object.
+ * @param {Object} bidderRequest - The bidder request object.
+ * @returns {Object|null} The OpenRTB 2.5 request payload or null if missing mandatory parameters.
+ */
+function buildOpenRtbRequest(bid, bidderRequest) {
+  if (!bid || !bidderRequest) {
+    logError(`${LOG_ERROR_PREFIX} Missing required parameters for OpenRTB request.`);
+    return null;
+  }
+
+  const imp = [];
+
+  // Handle different media types (Banner, Video, Native)
+  if (bid.mediaTypes.banner) {
+    imp.push({
+      id: bid.bidId,
+      banner: {
+        format: bid.sizes.map(size => ({ w: size[0], h: size[1] })), // List of size objects
+        w: bid.sizes[0][0],
+        h: bid.sizes[0][1],
+      },
+      bidfloor: bid.params.bidfloor || 0, // Handle bid floor if specified
+      secure: isSecureRequest() ? 1 : 0, // Indicates whether the request is secure (HTTPS)
+    });
+  }
+
+  if (bid.mediaTypes.video) {
+    imp.push({
+      id: bid.bidId,
+      video: {
+        w: bid.sizes[0][0],
+        h: bid.sizes[0][1],
+        mimes: bid.mediaTypes.video.mimes || ['video/mp4'], // Default to video/mp4 if not specified
+        protocols: bid.mediaTypes.video.protocols || [2, 3, 5, 6], // RTB video ad serving protocols
+        maxduration: bid.mediaTypes.video.maxduration || 30,
+        linearity: bid.mediaTypes.video.linearity || 1,
+        playbackmethod: bid.mediaTypes.video.playbackmethod || [2],
+      },
+      bidfloor: bid.params.bidfloor || 0,
+      secure: isSecureRequest() ? 1 : 0, // Indicates whether the request is secure (HTTPS)
+    });
+  }
+
+  if (bid.mediaTypes.native) {
+    imp.push({
+      id: bid.bidId,
+      native: {
+        request: JSON.stringify(bid.mediaTypes.native), // Convert native request to JSON string
+      },
+      bidfloor: bid.params.bidfloor || 0,
+      secure: isSecureRequest() ? 1 : 0, // Indicates whether the request is secure (HTTPS)
+    });
+  }
+
+  // Construct the OpenRTB request object
+  const openRtbRequest = {
+    id: bidderRequest.auctionId,
+    imp: imp,
+    site: {
+      page: bidderRequest.refererInfo.page,
+      domain: bidderRequest.refererInfo.domain,
+      ref: bidderRequest.refererInfo.ref || '', // Referrer URL
+    },
+    device: {
+      ua: navigator.userAgent,
+      ip: bidderRequest.ip || '123.123.123.123', // IP address (replace with actual)
+      devicetype: getDeviceType(), // 1 = Mobile/Tablet, 2 = Desktop
+      os: getDeviceOS(),
+      osv: getDeviceOSVersion(),
+      make: navigator.vendor || '',
+      model: getDeviceModel(),
+      connectiontype: getConnectionType(), // Include connection type
+      geo: {
```

Comment:
Get this from the request

Reply:
Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

Reply:
Done.